apps/core/src/auth/cognito-jwt-auth.guard.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { ExecutionContext, UnauthorizedException } from '@nestjs/common';
+import { HttpArgumentsHost } from '@nestjs/common/interfaces';
+import { localCognitoJwtVerifier } from '../config/local-cognito-jwt-verifier';
+import { CognitoJwtAuthGuard } from './cognito-jwt-auth.guard';
+
+const constructExecutionContext = (headers?: { authorization?: string }) => {
+  const notImplementedFunction = jest.fn(() => {
+    throw new Error('Function not implemented.');
+  });
+  const httpArgumentsHost: HttpArgumentsHost = {
+    getRequest: <T>(): T => ({ headers } as T),
+    getResponse: notImplementedFunction,
+    getNext: notImplementedFunction,
+  };
+  const executionContext: ExecutionContext = {
+    getClass: <T>(): T => Object as T,
+    getHandler: notImplementedFunction,
+    getArgs: notImplementedFunction,
+    getArgByIndex: notImplementedFunction,
+    switchToRpc: notImplementedFunction,
+    switchToHttp: jest.fn(() => httpArgumentsHost),
+    switchToWs: notImplementedFunction,
+    getType: notImplementedFunction,
+  };
```

Comment:
Does NestJS not provide test utils for automocking these?

Reply:
I don't think so. However, there is this lib [@golevelup/ts-jest](https://www.npmjs.com/package/@golevelup/ts-jest) that does. I wasn't ready for the commitment. Give me some time to consider it